src/OAuth2/ResponseType/JwtAccessToken.php

```diff
@@ -127,5 +127,21 @@ protected function createPayload($client_id, $user_id, $scope = null)
             'token_type' => $this->config['token_type'],
             'scope'      => $scope
         );
+        
+        if (isset($this->config['jwt_extra_payload_callable'])) {
+            if (!is_callable($this->config['jwt_extra_payload_callable'])) {
+                throw new \InvalidArgumentException("config['jwt_extra_payload_callable'] is not callable");
+            }
+            
+            $extra = call_user_func($this->config['jwt_extra_payload_callable'], $client_id, $user_id, $scope);
```

Comment:
Why don't we just pass this function `$payload` and expect the returned array to be the new payload, e.g.:

```php
$payload = call_user_func($this->config['jwt_extra_payload_callable'], $payload);
```

Reply:
This may break proper work of JWT
See our conversation with @svycka - https://github.com/bshaffer/oauth2-server-php/pull/804/files/86bd64e745ee34fa03bfb3456e5d2a2ab0a1c07e#r98169436
I agree to replace this
`$payload = array_merge($extra, $payload);`
to this
`$payload = array_merge($payload, $extra);`

Reply:
then, we should also rename the parameter to `jwt_override_payload_callable`

Reply:
I'm convinced. Ty for the explanation.